Cookbook/Proposals/ManagingAssets.md

```diff
@@ -0,0 +1,111 @@
+``` 
+enum Asset {
+    static let close = ImageAsset(name: "Close") 
+}
+``` 
+Then you can use it like this: 
+``` let closeImage = UIImage(asset: Asset.close) ``` 
+or 
+``` let sameCloseImage = Asset.close.image ``` 
+
+### Usage 
+
+Ultimately we would like to have a solution which allows us just add new asset and be able to use it immediately: 
+1. To add new asset e.g. close icon to the project it just should be placed in `Assets.xcassets` catalog in `BabylonDependencies.framework` 
+2. And then just use it like e.g: `Asset.close.image` (Xcode will report an error that will be fixed by recompiling). 
+
+To achieve this level of simplicity we have to extend the `SwiftGen` configuration to create identifiers for assets. 
+
+### Structure
+ 
+We agreed to have a flat structure for assets which increases discoverability (the same approach was chosen for localization). 
+
+### Overriding 
+
+To support the overriding of standard icons in white label apps we should update `SwiftGen` implementation to firstly access the image from `Bundle.main` which will pick target specific image and then fallback to `BabylonDependencies` if the image was not overridden. By using `SwiftGen`'s' custom template we can provide implementation like this:
+```
+public struct ImageAsset {
+  public fileprivate(set) var name: String
+
+  public var image: AssetImageTypeAlias {
+    let bundle = Bundle(for: BundleToken.self)
+    let img = image(for: Bundle.main) ?? image(for: bundle)
+    guard let result = img else { fatalError("Unable to load image named \(name).") }
+    return result
+  }
+
+  private func image(for bundle: Bundle) -> AssetImageTypeAlias? {
+    #if os(iOS) || os(tvOS)
+    let image = AssetImageTypeAlias(named: name, in: bundle, compatibleWith: nil)
+    #elseif os(OSX)
+    let image = bundle.image(forResource: NSImage.Name(name))
+    #elseif os(watchOS)
+    let image = AssetImageTypeAlias(named: name)
+    #endif
+    return image
+  }
+}
+```
+
+We should also configure SwiftGen to generate identifiers for additional assets provided by application-specific targets. We can configure `SwiftGen` to name `enum`s like `enum TargetAsset` to not have conflicts with BabylonDependencies' `enum Asset`.
+
+### Other rules
+ 
+Besides standard rules of adding new assets we should also take care when: 
+1. We are updating an icon we should check if any other place uses it. If so and that other place shouldn't be updated we should create a new icon with an updated design. 
+2. If we are removing some code which is using some icons we should check if that code was the last place which was using a given icon, if so icon should be deleted from `Assets.xcassets`. This step could be automated: if we are not using a specific image its identifier won't be used across the project. We can write the script which is checking the usage of every asset and when it is not used anywhere the asset in `Assets.xcassets` the catalog can be deleted. 
```

Comment:
```suggestion
2. If we are removing some code which is using some icons we should check if that code was the last place which was using a given icon, if so icon should be deleted from `Assets.xcassets`. This step could be automated: if we are not using a specific image its identifier won't be used across the project. 
We can write a script which checks the usage of every asset and when it is not used anywhere, it can be deleted. 
```